[Bug] Fix condition for disabling controls in product cards

samiuelson opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #10216 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR disables editable controls in order's product cards in case the order is not editable (e.g. it's in `completed` state, and the lock icon is visible).

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
1. Create order and set it's state to `completed`
2. Open the order and click `Edit`
3. Verify you it's impossible to edit products

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
<img src=https://github.com/woocommerce/woocommerce-android/assets/4527432/4bece057-5f7e-4fd6-a200-a86a9d96d24e width=300/>

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

samiuelson committed:
Fix condition for disabling controls in product cards

samiuelson committed:
Update release notes

samiuelson committed:
Merge branch 'trunk' into 10216-disable-editable-controls-when-orderd-not-editable

samiuelson committed:
Fix detekt issues

kidinov commented on RELEASE-NOTES.txt:
```diff
@@ -6,6 +6,7 @@
 -----
 - [**] Introduce ability to start creating a product by sharing images from other apps [https://github.com/woocommerce/woocommerce-android/pull/10331]
 - [*] [Internal] Improvements to the performance and internal logic of the barcode scanning screen [https://github.com/woocommerce/woocommerce-android/pull/10307]
+- [*] [Internal] Fixed disabling editable controls in product cards in order edit screen in case order is not editable [https://github.com/woocommerce/woocommerce-android/pull/10345]
```
That's 16.6, but the PR goes to 16.7 according to the milestone

samiuelson replied:
Fixed – d83bcd86b77ae068dee1c908e4280d1c05f6a63d

kidinov commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt:
```diff
@@ -620,7 +620,8 @@ private fun AmountPicker(
                     .widthIn(min = 12.dp, max = 128.dp)
-                    .width(IntrinsicSize.Min)
+                    .width(IntrinsicSize.Min),
+                enabled = isAmountChangeable,
```
Just trying to understand the goal here. We already have `readOnly = !isAmountChangeable,` which is kinda the same but the field is still focusable, right? I am wondering if you want to make it completely disabled, do we need `readOnly` property to be set then?

samiuelson replied:
Without `enabled` there is the `textStyle` animation is enabled when tapping the text field

https://github.com/woocommerce/woocommerce-android/assets/4527432/12a81a51-831d-4136-a504-844faba2d095

I replaced `readOnly` with `enabled` – d9c187a8bf986c0992197806c5c90a40abe1e80e

kidinov reviewed (approved):
LGTM. I left one kinda blocking comment regarding the release notes, and one question

samiuelson committed:
Fix release notes entry

samiuelson committed:
Replace TextField's `readOnly` with `enabled`

samiuelson committed:
Merge branch 'trunk' into 10216-disable-editable-controls-when-orderd-not-editable

samiuelson commented:
> LGTM. I left one kinda blocking comment regarding the release notes, and one question

Thanks for review @kidinov! I've addressed your comments 👍🏼

samiuelson committed:
Merge branch `trunk`

samiuelson committed:
Address detekt issues
